woocommerce/woocommerce-android
# Fixes disappearance of the "collect payment" button on the Order detail screen

Closes: #5048 


### Description

This PR fixes the disappearance of the "collect payment" button on the Order detail screen when the payment fails for any reason

Here is what was happening before this PR
1. Make the payment fail. For example, create any order with decimal quantity (#5048 has more detail on how to do this)
2. Try to collect payment from the order detail screen
3. Payment fails
4. Now, try to refresh the order detail screen
5. Collect payment button was disappearing

Because backend sends paymentMethod as "woocommerce_payments" after payment failure. `CardReaderPaymentCollectibilityChecker.kt` does not know how to handle this payment method and thus, it makes this order in-eligible for IPP.  Thus, it hides the collect payment button

This PR fixes it

### Simulating the Card reader
1. Go to the app's `build.gradle` file
2. Replace `project.hasProperty("useSimulatedReader").toString()` with `true`

### Criteria for a store and an order to be eligible for IPP
1. Store must have WCPay plugin installed 
2. Store's default address has to be a USA address
3. The order has to be COD

PdfdoF-V-p2 Here is an In-person payments FAQ that has more info on setting up the store and everything related to IPP. 

If you have any questions. Hit me up!

### Testing instructions

1. Create any order with decimal quantity
2. Try to collect payment from the order detail screen
3. Payment fails
4. Now, try to refresh the order detail screen
5. Make sure that the Collect payment button is showing
